src/services/pact-verifier.service.ts

```diff
@@ -12,19 +11,12 @@ export class PactVerifierService {
   ) {}
 
   public async verify(app: INestApplication): Promise<any> {
-    const host = this.options.providerHost || 'localhost';
-    await app.listen(await getPort(), host);
+    const providerUrl = new URL(this.options.providerBaseUrl);
```

Comment:
Ohh, I see what's going on here.

So, normally in Pact, the `providerBaseUrl` points to the (already running) provider at verification time. In nestjs-pact, the `providerHost` (which is changed back to `providerBaseUrl` here) is the host to start the provider on.

One advantage of `providerHost` is that it's not the same name as `providerBaseUrl`, and so not confused with the same named parameter (which has a different meaning) from Pact. One disadvantage is that it's named closely enough to be confusing. Something like `startProviderAtHost` is probably better.

In any case, this is a breaking change, since this PR has essentially renamed the parameter, and made it not optional by removing the default to `'localhost'`. If left as is, it should definitely be documented (also I think this is the only use of `getPort`, so if you leave it this way you can remove the dependency too).

However, I think it would be better to leave it as it was, with the `providerHost` parameter.

(I do like the shorter form of this function, though - inlining the port and hostname is a nice style improvement)